src/cinder/Signals.cpp

```diff
@@ -111,6 +111,22 @@ ScopedConnection& ScopedConnection::operator=( ScopedConnection &&rhs )
 	return *this;
 }
 
+ConnectionList::~ConnectionList()
+{
+	clear();
+}
+
+void ConnectionList::add( Connection &&target )
+{
+	mConnections.emplace_back( target );
+}
+
+void ConnectionList::clear()
+{
+	for( auto &conn : mConnections )
```

Comment:
Actually, ScopedConnection look pretty broken to me: It derives from noncopyable, but then defines a copy constructo/assignment operator. The resulting semantics is that whichever copy is destructed first, will disconnect the link. Also: If the move constructor / assignment operator is not marked as noexcept, std::vector will copy and not move it's elements when reallocating.

Together, this would probably explain the problems you had in the past.